libkbfs/errors.go

```diff
@@ -741,19 +741,43 @@ func (e MDServerDisconnected) Error() string {
 	return "MDServer is disconnected"
 }
 
-// MDUpdateApplyError indicates that we tried to apply a revision that
+// MDRevisionMismatch indicates that we tried to apply a revision that
 // was not the next in line.
-type MDUpdateApplyError struct {
+type MDRevisionMismatch struct {
 	rev  MetadataRevision
 	curr MetadataRevision
 }
 
-// Error implements the error interface for MDUpdateApplyError.
-func (e MDUpdateApplyError) Error() string {
+// Error implements the error interface for MDRevisionMismatch.
+func (e MDRevisionMismatch) Error() string {
 	return fmt.Sprintf("MD revision %d isn't next in line for our "+
 		"current revision %d", e.rev, e.curr)
 }
 
+// MDTlfIDMismatch indicates that the ID field of a successor MD
+// doesn't match the ID field of its predecessor.
+type MDTlfIDMismatch struct {
+	currID TlfID
+	nextID TlfID
+}
+
+func (e MDTlfIDMismatch) Error() string {
+	return fmt.Sprintf("TLF ID %s doesn't match successor TLF ID %s",
+		e.currID, e.nextID)
+}
+
+// MDPrevRootMismatch indicates that the PrevRoot field of a successor
+// MD doesn't match the ID of its predecessor.
```

Comment:
On 06/24/2016 08:57 PM, Fred Akalin wrote:
>
> In libkbfs/errors.go
> <https://github.com/keybase/kbfs/pull/68#discussion_r68483593>:
>
>
> > +}
> > +
> > +// MDPrevRootMismatch indicates that the PrevRoot field of a successor
> > +// MD doesn't match the ID of its predecessor.
>
> But it's the actual MD ID of the predecessor, right? Changed to
> "Metadata ID"
>

Oops, you're right!  Yeah, that's more clear.